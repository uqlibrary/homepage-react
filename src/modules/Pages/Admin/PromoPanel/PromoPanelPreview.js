/* eslint-disable no-unused-vars */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Dialog from '@material-ui/core/Dialog';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';
import parse from 'html-react-parser';
import Button from '@material-ui/core/Button';

const moment = require('moment');

import { StandardCard } from 'modules/SharedComponents/Toolbox/StandardCard';

const useStyles = makeStyles(theme => ({
    contentBox: {
        minWidth: '90%',
        paddingTop: 20,
        '& img': {
            maxWidth: 800,
            height: 800,
            border: '1px solid grey',
            textAlign: 'center',
        },
        '& li': {
            marginBottom: 10,
            padding: 10,
            '&:hover': {
                backgroundColor: theme.palette.secondary.main,
                transition: 'background-color 1s ease',
            },
            '& p': {
                marginBottom: 0,
                marginTop: 1,
            },
        },
        '& [aria-labelledby="lightboxTitle"]': {
            color: 'blue',
        },
    },
    dialogPaper: {
        // make the block take up more of the page
        width: 400,
    },
    link: {
        marginBottom: 10,
        marginRight: 10,
        cursor: 'pointer',
    },
}));
export const PromoPanelPreview = props => {
    const classes = useStyles();
    console.log('PROPS for the object', props);

    return (
        <React.Fragment>
            <Dialog
                open={props.isPreviewOpen}
                aria-labelledby="lightboxTitle"
                PaperProps={{ classes: { root: classes.dialogPaper } }}
            >
                <DialogTitle
                    id="lightboxTitle"
                    data-testid="spotlights-viewbyimage-lightbox-title"
                    style={{ position: 'relative', borderBottom: '1px solid #d7d1cc', fontSize: 12 }}
                    children={
                        <>
                            <p style={{ lineHeight: 1, margin: 0 }}>{'Preview'}</p>
                            <p style={{ fontSize: 14, fontWeight: 100, lineHeight: 1, margin: 0 }}>
                                {props.previewName}
                            </p>
                            <p style={{ fontSize: 12, fontWeight: 100 }}>
                                <b>{'For Group: '}</b>
                                {props.previewGroup.join(', ')}
                            </p>
                            {!!props.previewScheduled && (
                                <p style={{ fontSize: 12, fontWeight: 100 }}>
                                    <b>Schedule to</b>
                                    <br />
                                    <b>Start:</b> {moment(props.previewStart).format('dddd DD/MM/YYYY HH:mm a')}
                                    <br />
                                    <b>End:</b> {moment(props.previewEnd).format('dddd DD/MM/YYYY HH:mm a')}
                                </p>
                            )}
                        </>
                    }
                />
                <DialogContent>
                    <StandardCard
                        primaryHeader
                        fullHeight
                        standardCardId="promo-panel"
                        title={
                            <Grid container>
                                <Grid item xs={10} style={{ overflow: 'hidden', textOverflow: 'ellipsis' }}>
                                    {props.previewTitle}
                                </Grid>
                            </Grid>
                        }
                    >
                        <Grid container spacing={1}>
                            <Grid item xs>
                                {parse(props.previewContent)}
                            </Grid>
                        </Grid>
                    </StandardCard>
                    <Grid item xs={12} align="right">
                        <Button
                            style={{ marginTop: 10 }}
                            color="secondary"
                            children="Close"
                            data-testid="admin-promopanel-form-button-cancel"
                            variant="contained"
                            onClick={props.handlePreviewClose}
                        />
                    </Grid>
                </DialogContent>
            </Dialog>
        </React.Fragment>
    );
};

PromoPanelPreview.propTypes = {
    isPreviewOpen: PropTypes.bool,
    previewName: PropTypes.string,
    previewTitle: PropTypes.string,
    previewContent: PropTypes.string,
    previewGroup: PropTypes.array,
    previewScheduled: PropTypes.bool,
    previewStart: PropTypes.string,
    previewEnd: PropTypes.string,
    handlePreviewClose: PropTypes.func,
    handleLightboxClose: PropTypes.func,
    helpButtonLabel: PropTypes.string,
    helpContent: PropTypes.any,
    isLightboxOpen: PropTypes.bool,
    showViewByHistoryLightbox: PropTypes.func,
    spotlightImageUrl: PropTypes.string,
    spotlights: PropTypes.array,
};

PromoPanelPreview.defaultProps = {
    previewGroup: [],
    previewContent: '',
    helpButtonLabel: 'Help',
    helpContent: 'test',
    spotlights: [],
};

export default React.memo(PromoPanelPreview);
