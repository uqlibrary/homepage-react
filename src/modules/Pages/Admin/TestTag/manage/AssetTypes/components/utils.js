export const emptyActionState = { isAdd: false, isEdit: false, isDelete: false, rows: {}, row: {}, title: '' };

export const actionReducer = (_, action) => {
    switch (action.type) {
        case 'add':
            return {
                title: 'Add Asset Type',
                isAdd: true,
                isEdit: false,
                isDelete: false,
                row: { asset_type_id: 'auto' },
            };
        case 'edit':
            return { title: 'Edit Asset Type', isAdd: false, isEdit: true, isDelete: false, row: action.row };
        case 'clear':
            return { ...emptyActionState };
        case 'delete':
            return { isAdd: false, isEdit: false, isDelete: true, row: action.row };
        default:
            throw `Unknown action '${action.type}'`;
    }
};
