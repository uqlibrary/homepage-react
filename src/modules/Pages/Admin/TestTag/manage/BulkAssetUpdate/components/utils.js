import { createLocationString } from '../../../helpers/helpers';

export const transformRow = row => {
    return row.map(line => {
        if (!!line?.asset_location) return line;
        return {
            ...line,
            asset_type_name: line?.asset_type?.asset_type_name ?? '',
            asset_location: !!line?.last_location
                ? createLocationString({
                      site: line.last_location.site_name,
                      building: line.last_location.building_name,
                      floor: line.last_location.floor_id_displayed,
                      room: line.last_location.room_id_displayed,
                  })
                : '',
        };
    });
};

export const transformRequest = formValues => {
    if (!!formValues.hasDiscardStatus) {
        formValues.hasLocation = false;
        formValues.hasAssetType = false;
    } else {
        formValues.hasDiscardStatus = false;
    }
    return {
        asset: formValues.asset_list.reduce((cumulative, current) => [...cumulative, current.asset_id], []),
        ...(!!formValues.hasLocation ? { asset_room_id_last_seen: formValues.location.room } : {}),
        ...(!!formValues.hasAssetType ? { asset_type_id: formValues.asset_type.asset_type_id } : {}),
        ...(!!formValues.hasDiscardStatus ? { is_discarding: 1, discard_reason: formValues.discard_reason } : {}),
    };
};

export const transformFilterRow = row => {
    return row.map(line => {
        if (!!line?.asset_id_displayed) return line;
        return {
            ...line,
            asset_id_displayed: line?.asset_barcode ?? '',
            asset_location: createLocationString({
                site: line.site_name,
                building: line.building_name,
                floor: line.floor_id_displayed,
                room: line.room_id_displayed,
            }),
        };
    });
};
