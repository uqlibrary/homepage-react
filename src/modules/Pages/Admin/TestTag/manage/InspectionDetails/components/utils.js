import moment from 'moment';
import locale from '../../../testTag.locale';
const dateFormat = locale.config.format.dateFormatNoTime;

export const transformRow = row => {
    return row.map(line => {
        if (!!row.user_name) return row;
        return {
            asset_id: line.asset_id,
            asset_id_displayed: line.asset_id_displayed,
            asset_type: line.asset_type?.asset_type_name,
            asset_status: line.asset_status,
            user_name: line.last_inspection?.user_name,
            inspect_date: !!line.last_inspection ? moment(line.last_inspection.inspect_date).format(dateFormat) : '',
            inspect_notes: line.last_inspection?.inspect_notes ?? '',
            inspect_fail_reason: line.last_inspection?.inspect_fail_reason ?? '',
            discard_reason: line.last_discard?.discard_reason ?? '',
            repairer_name: line.last_repair?.repairer_name ?? '',
            last_inspect_status: line.last_inspection?.inspect_status ?? '',
        };
    });
};

export const transformUpdateRequest = request => {
    const clonedRequest = structuredClone(request);
    const { asset_status: assetStatus, last_inspect_status: lastInspectStatus } = clonedRequest;
    const defaultValues = { inspect_notes: clonedRequest.inspect_notes };

    switch (assetStatus) {
        case locale.config.assetStatus.current:
            return defaultValues;
        case locale.config.assetStatus.failed:
            return { ...defaultValues, inspect_fail_reason: clonedRequest.inspect_fail_reason };
        case locale.config.assetStatus.outforrepair:
            return {
                ...defaultValues,
                inspect_fail_reason: clonedRequest.inspect_fail_reason,
                repairer_name: clonedRequest.repairer_name,
            };
        case locale.config.assetStatus.discarded:
            return {
                ...defaultValues,
                ...(lastInspectStatus === locale.config.inspectStatus.failed
                    ? { inspect_fail_reason: clonedRequest.inspect_fail_reason }
                    : {}),
                discard_reason: clonedRequest.discard_reason,
            };
        default:
            return request;
    }
};
