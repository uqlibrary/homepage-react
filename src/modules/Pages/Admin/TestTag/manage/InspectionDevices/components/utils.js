import moment from 'moment';
import locale from 'modules/Pages/Admin/TestTag/testTag.locale';
const dateFormat = locale.pages.manage.config.dateFormat;

export const emptyActionState = { isAdd: false, isEdit: false, isDelete: false, title: '', row: {} };
export const actionReducer = (_, action) => {
    const { type, row, title, ...props } = action;
    switch (type) {
        case 'add':
            return {
                isAdd: true,
                isEdit: false,
                isDelete: false,
                row: {
                    device_id: 'auto',
                    device_calibrated_date_last: moment().format(dateFormat),
                    device_calibration_due_date: moment()
                        .add(1, 'd')
                        .format(dateFormat),
                },
                title,
                props: { ...props },
            };
        case 'edit':
            return { isAdd: false, isEdit: true, isDelete: false, title, row, props: { ...props } };
        case 'delete':
            return { isAdd: false, isEdit: false, isDelete: true, title, row, props: { ...props } };
        case 'clear':
            return { ...emptyActionState };
        default:
            throw `Unknown action '${type}'`;
    }
};

export const formatDateStrings = (row, suffix) => {
    const dateLastParts = row.device_calibrated_date_last.split(' ');
    const dateNextParts = row.device_calibration_due_date.split(' ');
    return {
        ...row,
        device_calibrated_date_last:
            dateLastParts.length > 1 ? row.device_calibrated_date_last : `${row.device_calibrated_date_last} ${suffix}`,
        device_calibration_due_date:
            dateNextParts.length > 1 ? row.device_calibration_due_date : `${row.device_calibration_due_date} ${suffix}`,
    };
};

export const transformAddRequest = (request, user) => {
    delete request.device_id;
    delete request.device_current_flag;

    const newResponse = { ...formatDateStrings(request, '00:00:00'), device_department: user.user_department };
    return newResponse;
};

export const transformUpdateRequest = request => {
    delete request.device_current_flag;

    const newResponse = formatDateStrings(request, '00:00:00');
    return newResponse;
};

export const transformRow = row => {
    return row.map(line => ({
        ...line,
        device_calibrated_date_last: moment(line.device_calibrated_date_last).format(
            locale.config.format.dateFormatNoTime,
        ),
        device_calibration_due_date: moment(line.device_calibration_due_date).format(
            locale.config.format.dateFormatNoTime,
        ),
    }));
};
