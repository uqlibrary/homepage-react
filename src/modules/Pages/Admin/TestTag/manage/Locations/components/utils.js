import React from 'react';
import RowMenuCell from './../../../SharedComponents/DataTable/RowMenuCell';

export const createLocationString = ({ site, building, floor }) =>
    [site, building, floor].filter(item => !!item).join(' / ');

export const getColumns = ({ config, locale, selectedFilter, handleEditClick, handleDeleteClick }) => {
    const actionsCell = {
        field: 'actions',
        headerName: locale?.actions,
        renderCell: params => {
            return (
                <RowMenuCell
                    {...params}
                    handleEditClick={handleEditClick}
                    {...((params.row?.asset_count ?? 1) === 0 ? { handleDeleteClick: handleDeleteClick } : {})}
                />
            );
        },
        sortable: false,
        width: 100,
        headerAlign: 'center',
        filterable: false,
        align: 'center',
        disableColumnMenu: true,
        disableReorder: true,
        renderInUpdate: false,
    };

    const columns = [];
    const keys = Object.keys(config[selectedFilter].fields);

    keys.forEach(key => {
        !!(config[selectedFilter].fields[key]?.fieldParams.renderInTable ?? true) &&
            columns.push({
                field: key,
                headerName: locale?.[selectedFilter]?.[key].label,
                editable: false,
                sortable: false,
                ...config[selectedFilter].fields[key].fieldParams,
            });
    });

    columns && columns.length > 0 && columns.push(actionsCell);
    return columns;
};

export const emptyActionState = { isAdd: false, isEdit: false, isDelete: false, row: {} };
export const actionReducer = (_, action) => {
    const { type, row, selectedFilter, ...props } = action;
    console.log('actionReducer', type, props);
    switch (type) {
        case 'add':
            return {
                isAdd: true,
                isEdit: false,
                isDelete: false,
                row: { [`${selectedFilter}_id`]: 'auto' },
                props: { ...props },
            };
        case 'edit':
            return { isAdd: false, isEdit: true, isDelete: false, row, props: { ...props } };
        case 'delete':
            return { isAdd: false, isEdit: false, isDelete: true, row, props: { ...props, selectedFilter } };
        case 'clear':
            return { ...emptyActionState };
        default:
            throw `Unknown action '${type}'`;
    }
};

export const transformAddRequest = ({ request, selectedFilter, location }) => {
    // add requests may have an id field (probably 'auto') that needs to be removed
    delete request[`${selectedFilter}_id`];

    // get location key names i.e. ['site','building','floor','room']
    const keys = Object.keys(location);

    // get current selected location index
    const index = keys.findIndex(key => key === selectedFilter);

    // index 0 (sites) doesnt have a preceding location, so just return the request
    if (index === 0) return request;

    // else grab the name of the previous location type i.e. if selectedFilter is 'room', this would return 'building'
    const prevKey = keys[index - 1];

    // build a new request by inserting a key:value in format e.g. 'room_floor_id: 1'
    return { ...request, [`${selectedFilter}_${prevKey}_id`]: location[prevKey] };
};

export const transformUpdateRequest = ({ request, selectedFilter, location }) => {
    // get location key names i.e. ['site','building','floor','room']
    const keys = Object.keys(location);

    // get current selected location index
    const index = keys.findIndex(key => key === selectedFilter);

    // index max (room) doesnt have any superseding locations, so just return the request
    if (index === keys.length - 1) return request;

    // else grab the name of the next location type i.e. if selectedFilter is 'building', this would return 'room'
    const nextKey = keys[index + 1];

    // now remove the array of next location data we got from the server e.g.
    // for sites, we also get a buildings:[] array (note plural), for buildings
    // we also get a rooms:[] array etc.
    // We dont need to send all of this stuff back to the server.
    delete request[`${nextKey}s`];

    return request;
};
