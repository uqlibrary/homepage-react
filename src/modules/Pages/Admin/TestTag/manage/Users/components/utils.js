export const transformRow = row => {
    return row.map(line => ({
        ...line,
        can_admin_cb: line?.privileges?.can_admin === 1,
        can_inspect_cb: line?.privileges?.can_inspect === 1,
        can_alter_cb: line?.privileges?.can_alter === 1,
        can_see_reports_cb: line?.privileges?.can_see_reports === 1,
        user_current_flag_cb: line?.user_current_flag === 1,
        can_admin: line?.privileges?.can_admin === 1 ? 'Yes' : 'No',
        can_inspect: line?.privileges?.can_inspect === 1 ? 'Yes' : 'No',
        can_alter: line?.privileges?.can_alter === 1 ? 'Yes' : 'No',
        can_see_reports: line?.privileges?.can_see_reports === 1 ? 'Yes' : 'No',
        user_current_flag: line?.user_current_flag === 1 ? 'Yes' : 'No',
    }));
};

export const emptyActionState = { isAdd: false, isEdit: false, isDelete: false, title: '', row: {} };

export const transformUpdateRequest = request => {
    // delete request.previleges;
    delete request.inspectionCount;
    delete request.id;
    request.privileges.can_admin = request?.can_admin_cb ? 1 : 0;
    request.privileges.can_inspect = request?.can_inspect_cb ? 1 : 0;
    request.privileges.can_alter = request?.can_alter_cb ? 1 : 0;
    request.privileges.can_see_reports = request?.can_see_reports_cb ? 1 : 0;

    request.user_current_flag = request?.user_current_flag_cb ? 1 : 0;

    delete request.can_admin;
    delete request.can_admin_cb;
    delete request.can_inspect;
    delete request.can_inspect_cb;
    delete request.can_alter;
    delete request.can_alter_cb;
    delete request.can_see_reports;
    delete request.can_see_reports_cb;

    return request;
};

export const transformAddRequest = request => {
    // delete request.previleges;
    console.log('REquest is ', request);
    delete request.inspectionCount;
    delete request.id;

    request.privileges.can_admin = request?.can_admin_cb ? 1 : 0;
    request.privileges.can_inspect = request?.can_inspect_cb ? 1 : 0;
    request.privileges.can_alter = request?.can_alter_cb ? 1 : 0;
    request.privileges.can_see_reports = request?.can_see_reports_cb ? 1 : 0;

    request.user_current_flag = request?.user_current_flag_cb ? 1 : 0;

    delete request.can_admin;
    delete request.can_admin_cb;
    delete request.can_inspect;
    delete request.can_inspect_cb;
    delete request.can_alter;
    delete request.can_alter_cb;
    delete request.can_see_reports;
    delete request.can_see_reports_cb;

    return request;
};

export const actionReducer = (_, action) => {
    const { type, row, title, ...props } = action;
    switch (type) {
        case 'add':
            return {
                isAdd: true,
                isEdit: false,
                isDelete: false,
                row: {
                    user_id: 'Auto',
                    user_uid: '',
                    user_name: '',
                    user_current_flag_cb: true,
                },
                title,
                props: { ...props },
            };
        case 'edit':
            console.log('Edit Row', row);
            return {
                isAdd: false,
                isEdit: true,
                isDelete: false,
                title,
                row: { ...row, id: row.user_id },
                // row,
                props: { ...props },
            };
        case 'delete':
            return { isAdd: false, isEdit: false, isDelete: true, title, row, props: { ...props } };
        case 'clear':
            return { ...emptyActionState };
        default:
            throw `Unknown action '${type}'`;
    }
};
