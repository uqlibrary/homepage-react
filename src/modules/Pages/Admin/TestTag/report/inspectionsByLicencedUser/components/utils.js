import locale from '../../../testTag.locale';
const moment = require('moment');

export const getNameStyles = (name, inspectorName, theme) => ({
    fontWeight:
        inspectorName.indexOf(name) === -1 ? theme.typography.fontWeightRegular : theme.typography.fontWeightMedium,
});

export const transformRow = row => {
    return row.map(line => ({
        ...line,
        start_date: !!line?.start_date ? moment(line.start_date).format(locale.config.format.dateFormatNoTime) : '--',
        end_date: !!line?.start_date ? moment(line.end_date).format(locale.config.format.dateFormatNoTime) : '--',
    }));
};
