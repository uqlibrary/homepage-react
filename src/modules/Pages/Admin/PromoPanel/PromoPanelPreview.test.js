import { PromoPanelPreview } from './PromoPanelPreview';
import React from 'react';
import { rtlRender, fireEvent } from '../../../../../utils/test-utils';

function setup(testProps = {}) {
    const props = {
        isPreviewOpen: true,
        previewName: 'test preview',
        previewTitle: 'Title Preview',
        previewContent: 'Content Preview',
        handlePreviewClose: jest.fn(),
    };

    return rtlRender(<PromoPanelPreview {...props} {...testProps} />);
}

describe('Component PromoPanel Preview', () => {
    it('should render correctly', () => {
        const wrapper = setup();

        expect(wrapper).toMatchSnapshot();
    });
    it('should handle close correctly', () => {
        const closeFn = jest.fn();
        const { getByTestId } = setup({ handlePreviewClose: closeFn });

        fireEvent.click(getByTestId('admin-promopanel-preview-button-cancel'));
        expect(closeFn).toHaveBeenCalledTimes(1);
    });
});
