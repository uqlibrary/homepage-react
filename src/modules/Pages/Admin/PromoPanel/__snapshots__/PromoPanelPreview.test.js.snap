// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PromoPanel Preview should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby="promopanel-preview-title"
          class="MuiPaper-root makeStyles-dialogPaper-2 MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-1qg91f3-MuiTypography-root-MuiDialogTitle-root"
            data-testid="promopanel-preview-title"
            id="promopanel-preview-title"
            style="position: relative; border-bottom: 1px solid #d7d1cc; font-size: 12px;"
          >
            <p
              style="line-height: 1; margin: 0px;"
            >
              Preview
            </p>
            <p
              style="font-size: 14px; font-weight: 100; line-height: 1; margin: 0px;"
            >
              test preview
            </p>
          </h2>
          <div
            class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root Cards-card-4 StandardCard  css-bhp9pd-MuiPaper-root-MuiCard-root"
              data-analyticsid="promo-panel"
              data-testid="promo-panel"
              id="promo-panel"
              style="height: 100%;"
            >
              <div
                class="MuiCardHeader-root Cards-cardHeaderPrimary-6 css-185gdzj-MuiCardHeader-root"
                style="border-top-left-radius: 4px; border-top-right-radius: 4px;"
              >
                <div
                  class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
                >
                  <h2
                    class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1hzv55z-MuiTypography-root"
                    data-testid="promo-panel-header"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
                        style="overflow: hidden; text-overflow: ellipsis;"
                      >
                        Title Preview
                      </div>
                    </div>
                  </h2>
                </div>
                <div
                  class="MuiCardHeader-action css-sgoict-MuiCardHeader-action"
                />
              </div>
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
                data-testid="promo-panel-content"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
                  >
                    Content Preview
                  </div>
                </div>
              </div>
            </div>
            <div
              align="right"
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-13i4rnv-MuiGrid-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-da9nxa-MuiButtonBase-root-MuiButton-root"
                data-testid="admin-promopanel-preview-button-cancel"
                id="admin-promopanel-preview-button-cancel"
                style="margin-top: 10px;"
                tabindex="0"
                type="button"
              >
                Close
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
